create_derivatives/routines.py

```diff
@@ -65,8 +67,103 @@ def get_as_uri(self, bag_filepath):
 
 
 class JP2Maker:
-    # TO DO: make JPG2000 derivatives
-    pass
+    """Creates JP2000 derivatives from TIFFs.
+
+    Creates JP2 directory in bag's data directory and JP2 derivatives.
+    TIFFs to be converted are targeted in the service directory. If the service directory is empty or
+    does not exist, target TIFFs at the root of the data directory.
+
+    Returns:
+        A tuple containing human-readable message along with list of bag identifiers.
+        Exceptions are raised for errors along the way.
+    """
+
+    def run(self):
+        bags_with_jp2s = []
+        for bag in Bag.objects.filter(process_status=Bag.PREPARED):
+            service_dir = Path(bag.bag_path, "data", "service")
+            if service_dir.is_dir() and len(service_dir):
```

Comment:
why are we checking `len` here?

Reply:
Ah. So what I am trying to do is check if there is a service directory that is not empty. Should that be `len(service_dir) != 0`? Or am I thinking about that check incorrectly?

Reply:
What you're doing there is checking the length of the `Path` object, which is not the same thing as checking to see if the directory is empty :)

I think what you want is `any(service_dir.iterdir())`. Calling `iterdir` on a Path object returns an iterator function, so you can't just do `len(service_dir.iterdir())` - instead you need to use the `any` key, which evaluates to False if there's nothing to iterate over.